src/job_metadata.c

```diff
@@ -842,7 +845,13 @@ CronJobRelationId(void)
 	{
 		Oid cronSchemaId = get_namespace_oid(CRON_SCHEMA_NAME, false);
 
-		CachedCronJobRelationId = get_relname_relid(JOBS_TABLE_NAME, cronSchemaId);
+		Oid cronJobRelationId = get_relname_relid(JOBS_TABLE_NAME, cronSchemaId);
+
+		if (EnableCronJobRelationIdCaching)
```

Comment:
not a fan of this "function sometimes does a different thing depending on a global variable" behavior

Seems better to be explicit and have two functions. The cached one can call the uncached one and other code paths that don't need caching can as well.

Reply:
Made the change. But only 1 place calls the cached function and multiple places call the uncached function. So I would prefer to change the functions to `GetCronJobRelationId` and `CacheAndGetCronJobRelationId`. How do you feel about it?